tapdb/asset_minting.go

```diff
@@ -1470,6 +1470,70 @@ func (a *AssetMintingStore) CommitBatchTx(ctx context.Context,
 	})
 }
 
+// FetchDelegationKey fetches the delegation key for the given asset group
+// public key.
+func (a *AssetMintingStore) FetchDelegationKey(ctx context.Context,
+	groupKey btcec.PublicKey) (fn.Option[tapgarden.DelegationKey], error) {
+
+	var zero fn.Option[tapgarden.DelegationKey]
+	groupKeyBytes := groupKey.SerializeCompressed()
+
+	var delegationKey fn.Option[tapgarden.DelegationKey]
+
+	readOpts := NewAssetStoreReadTx()
+	dbErr := a.db.ExecTx(ctx, &readOpts, func(q PendingAssetStore) error {
+		fetchRow, err := q.FetchMintAnchorUniCommitment(
+			ctx, FetchPreCommitParams{
+				GroupKey: groupKeyBytes,
+			},
+		)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil
+			}
+			return fmt.Errorf("unable to fetch mint anchor "+
+				"uni commitment by group key: %w", err)
+		}
+
+		// If we didn't find any pre-commitment outputs, then
+		// we can return early.
+		if len(fetchRow) == 0 {
+			return nil
+		}
+
+		// If there are multiple pre-commitment outputs, just choose any
+		// single one, we assume they all have the same delegation key.
```

Comment:
Is this something we need to validate? Or was this validated at another point and we can make a more or less safe assumption here?

Reply:
Ah, I guess this is more or less what this whole commit is doing. So perhaps re-formulate this comment to "we choose the first one because we can assume all commitment outputs of the same group have the same delegation key"?